### PR TITLE
Fix bug where unauthorised register was retrying erroneously

### PR DESCRIPTION
**Pull request by moskyb**
### Description

A delicious, delicious single-character typo has caused unauthorised checkouts to retry for 5 minutes. There's no recovering from this error, so it shouldn't be retried.

### Testing
- [x] Tests have run locally (with `go test ./...`). Buildkite employees may check this if the pipeline has run automatically.
- [x] Code is formatted (with `go fmt ./...`)

### CODE REVIEW

File: agent/register.go [hunk 1]
```diff
@@ -53,15 +56,10 @@ func Register(ctx context.Context, l logger.Logger, ac APIClient, req api.AgentR
...
-		registered, resp = reg, rsp
-		return nil
-	}
 
-	// Try to register, retrying every 10 seconds for a maximum of 30 attempts (5 minutes)
-	err := roko.NewRetrier(
-		roko.WithMaxAttempts(30),
-		roko.WithStrategy(roko.Constant(10*time.Second)),
-	).DoWithContext(ctx, register)
+		registered = reg
```
Comment (triarius) [inline agent/register.go:60]: Does `registered` need to be different from `reg`?

**Review by triarius — APPROVED** [review]
Ouch, LGTM

File: agent/register.go (hunk 1 above)
Comment (moskyb) [reply agent/register.go:60]: i don't think it does! will change

**Review by triarius — APPROVED** [review]

**Review by DrJosh9000 — COMMENTED** [review]
Oh no! Thanks for fixing my mistake!

**Pull request merged**